### PR TITLE
`date_histogram` extended_bounds param

### PR DESCRIPTION
**Pull request by trzysiek**
Style could be improved, better abstractions to make code clearer, but not much time left today, and it seemes working (both tests + manual testing with Elastic), so I'd say it's good enough.

### CODE REVIEW

File: quesma/testdata/aggregation_requests.go [hunk 1]
```diff
@@ -3628,12 +3628,12 @@ var AggregationTests = []AggregationTestCase{
 			{Cols: []model.QueryResultCol{
 				model.NewQueryResultCol("aggr__sampler__count", uint64(15)),
 				model.NewQueryResultCol("aggr__sampler__eventRate__key_0", int64(1709816790000/15000)),
-				model.NewQueryResultCol("aggr__sampler__eventRate__count", 0),
+				model.NewQueryResultCol("aggr__sampler__eventRate__count", 2),
```
Comment (trzysiek) [inline quesma/testdata/aggregation_requests.go:3631]: Count 0 wouldn't be returned from Clickhouse, so I'm setting some proper value

File: quesma/model/bucket_aggregations/date_histogram.go [hunk 2]
```diff
@@ -286,17 +302,70 @@ func (qt *DateHistogramRowsTransformer) Transform(ctx context.Context, rowsFromD
...
+
+	// add "pre" keys, so any needed key between [extendedBoundsMin, first_row_key]
+	if qt.extendedBoundsMin != NoExtendedBound {
+		firstRequiredKey := qt.extendedBoundsMin/qt.differenceBetweenTwoNextKeys - 1 // more or less, might be slightly off, seems to work for a few different test cases
+		var lastRequiredKey int64
+		if len(postprocessedRows) > 0 {
+			lastRequiredKey = qt.getKey(postprocessedRows[0]) / qt.differenceBetweenTwoNextKeys
+		} else {
+			// we know qt.extendedBoundsMax != NoExtendedBound, because we would've returned earlier - line below is safe
+			lastRequiredKey = (qt.extendedBoundsMax + 1000*60*60*2) / qt.differenceBetweenTwoNextKeys
```
Comment (trzysiek) [inline quesma/model/bucket_aggregations/date_histogram.go:341]: This `1000*60*60*2)` needs to be changed (extracted via some timezone parameter passed in the request? I don't know yet, didn't start looking into it)

File: quesma/model/bucket_aggregations/date_histogram.go [hunk 3]
```diff
@@ -286,17 +307,74 @@ func (qt *DateHistogramRowsTransformer) Transform(ctx context.Context, rowsFromD
...
+
+		postprocessedRows = append(preRows, postprocessedRows...)
+	}
+
+	// add "post" keys, so any needed key between [last_row_key, extendedBoundsMax]
+	if qt.extendedBoundsMax != NoExtendedBound {
+		firstRequiredKey := qt.dateHistogram.calculateResponseKeyInUTC(qt.getKey(postprocessedRows[len(postprocessedRows)-1]))/qt.differenceBetweenTwoNextKeys + 1
+		lastRequiredKey := qt.dateHistogram.fromUTCToWantedTimezone(qt.extendedBoundsMax) / qt.differenceBetweenTwoNextKeys
+		for postKey := firstRequiredKey; postKey <= lastRequiredKey && emptyRowsAdded < maxEmptyBucketsAdded; postKey++ {
+			postRow := postprocessedRows[0].Copy()
```
Comment (jakozaur) [inline quesma/model/bucket_aggregations/date_histogram.go:370]: Make this logic for both bounds if element exists.

File: quesma/model/bucket_aggregations/date_histogram.go [hunk 4]
```diff
@@ -286,17 +307,80 @@ func (qt *DateHistogramRowsTransformer) Transform(ctx context.Context, rowsFromD
 				i-1, rowsFromDB[i-1], i, rowsFromDB[i],
 			)
 		}
-		lastKey := qt.getKey(rowsFromDB[i-1])
-		currentKey := qt.getKey(rowsFromDB[i])
+		lastKey := qt.dateHistogram.calculateResponseKeyInUTC(qt.getKey(rowsFromDB[i-1]))
+		currentKey := qt.dateHistogram.calculateResponseKeyInUTC(qt.getKey(rowsFromDB[i]))
+
+		// ugly, but works, will do for now
+		doWeDivide := (currentKey/qt.getKey(rowsFromDB[i])) >= 100 || (float64(currentKey)/float64(qt.getKey(rowsFromDB[i]))) <= 0.01
```
Comment (jakozaur) [inline quesma/model/bucket_aggregations/date_histogram.go:314]: This check means the value is almost changed. Not sure if the division is right.


```suggestion
                keyFromDb := qt.getKey(rowsFromDB[i])
		doWeDivide := (currentKey/keyFromDb) >= 100 || (float64(currentKey)/float64(keyFromDb)) <= 0.01
```

**Review by jakozaur — APPROVED** [review]

**Pull request merged**